### ISSUE TITLE
Issues with tensors shape with TensorFlow backend

### ISSUE THREAD

**Issue by alexattia** [fchollet/keras#7263]
I am trying to transform a TensorFlow script into Keras. I have a classification model which predicts the presence of digits and 4 digits. I have two different losses and two outputs: one for the presence and one for the digits.  

This is my model :

```
    def get_training_model2():
        inp, x = convolutional_layers2()
        x = Flatten()(x)
        x = Dense(2048)(x)
        x = Activation('relu')(x)
        y = Dense(1+5*len(DIGITS), activation='softmax')(x)
        out1 = Lambda(lambda a: a[:,:1])(y)
        out2 = Lambda(lambda a: a[:,1:])(y)
        model = Model(inp, [out1, out2])
        return model
```

This is my digit loss : 

```
    def digits_loss(y_true, y_pred):
        digits_loss = losses.sparse_categorical_crossentropy(K.reshape(y_true, [-1, len(DIGITS)]),
                                                             K.reshape(y_pred, [-1, len(DIGITS)]))
        digits_loss = np.sum(digits_loss)
        return digits_loss
    
    # Target    
    target = [y_train[:,:1],  y_train[:,1:]]
```

When I launch : 

```
    model = model_keras.get_training_model2()
    opt = Adam(lr=learn_rate)
    model.compile(loss=[presence_loss, digits_loss],
                  optimizer=opt,
                  metrics=['accuracy'])

    model.fit(X_train, target, batch_size=50,
              epochs=50 ...)
```
I have this error : 

```
    File "train_keras.py", line 96, in digits_loss
        K.reshape(y_pred, [-1, len(DIGITS)]))
    logits and labels must have the same first dimension, got logits shape [250,11] and labels shape [2750]
```

I don't understand why I have this error because I am reshaping both labels and logits in the same way. Moreover, I don't understand why the shape is [250,11] although it's should be [50, 11] (batch_size).  

This is the printed tensor shapes before passing them to `sparse_categorical_crossentropy`:

```
 y_true (?, ?) 
y_pred (?, 55) 
K.reshape(y_true, [-1, len(DIGITS)]) (?, 11) 
K.reshape(y_pred, [-1, len(DIGITS)]) (?, 11) 
```

Other shapes : 
```
X_train : (50, 64, 128, 1) 
y_train for presence loss: (50, 1) 
y_train for digits loss: (50, 55) 
```

Thank you for any help.
